django-backend/fecfiler/user/views.py

```diff
@@ -1,22 +1,31 @@
-from rest_framework import viewsets
-from rest_framework.decorators import action
-from .serializers import CurrentUserSerializer
-from django.http import HttpResponseBadRequest
 from rest_framework.response import Response
+from rest_framework.viewsets import ModelViewSet
+from .serializers import CurrentUserSerializer, session_security_consented_key
+from datetime import date, timedelta
 import logging
 
 logger = logging.getLogger(__name__)
 
 
-class UserViewSet(viewsets.GenericViewSet):
-    @action(detail=False, methods=["GET", "PUT"], serializer_class=CurrentUserSerializer)
-    def current(self, request):
-        if request.method == 'GET':
-            serializer = self.get_serializer(request.user)
-            return Response(serializer.data)
-        if request.method == 'PUT':
-            serializer = self.get_serializer(request.user, data=request.data)
-            if serializer.is_valid():
-                serializer.save()
-                return Response(serializer.data)
-            return HttpResponseBadRequest()
+class UserViewSet(ModelViewSet):
+
+    def retrieve(self, request, *args, **kwargs):
+        serializer = self.get_serializer(request.user)
+        return Response(serializer.data)
+
+    def update(self, request, *args, **kwargs):
```

Comment:
I think the front end is still expecting to work with `current`
![image](https://github.com/user-attachments/assets/b5a68ae7-a84d-456a-93d0-c8914c3a96e3)

Reply:
good catch, not sure how this slipped past my testing.  Updated the urls to get_current and update_current to avoid needing to set a user id in the url.